### ISSUE TITLE
Consistency training fails to converge

### ISSUE THREAD

**Issue by yhx-03** [huggingface/diffusers#8277]
**Consistency training fails to converge**
When I use examples/research_projects/consistency_training/train_cm_ct_unconditional.py to train the consistency model, I try to follow the experimental parameters of the original paper, but I cannot obtain normal generation results.

**Script**
```bash
python -m accelerate.commands.launch examples/research_projects/consistency_training/train_cm_ct_unconditional.py
--dataset_name="cifar10"
--dataset_image_column_name="img"
--output_dir="examples/research_projects/consistency_training/outputs_cifar10"
--resolution=32
--max_train_steps=400000
--dataloader_num_workers=8
--noise_precond_type="cm"
--input_precond_type="cm"
--train_batch_size=512
--gradient_accumulation_steps=2
--learning_rate=1e-04
--lr_scheduler="constant"
--lr_warmup_steps=0
--use_ema
--validation_steps=5000
--eval_batch_size=32
--checkpointing_steps=5000
--checkpoints_total_limit=10
--use_8bit_adam
--mixed_precision=fp16
--resume_from_checkpoint="latest"
```

**Generated results(step 400000)**
![step400000](https://github.com/huggingface/diffusers/assets/58554991/7b261e5e-c873-4462-97b7-808b2e471c38)

**Comment by bghira** [comment]
out of curiosity, have you tested sampling from the EMA vs non-EMA weights?

**Comment by yhx-03** [comment]
> out of curiosity, have you tested sampling from the EMA vs non-EMA weights?

results using non-EMA weights:
![image0](https://github.com/huggingface/diffusers/assets/58554991/046617a9-95ac-44c7-97c9-4fad72e8153a)
![image2](https://github.com/huggingface/diffusers/assets/58554991/4825c929-67ab-484f-9572-ec5b8bf3bb45)
![image3](https://github.com/huggingface/diffusers/assets/58554991/8fb6d7ff-c758-4535-a307-30e86b17b6b5)
![image4](https://github.com/huggingface/diffusers/assets/58554991/3446ba64-4e8f-4eef-8ac8-77e53574a2e1)
![image5](https://github.com/huggingface/diffusers/assets/58554991/99234e69-ce3a-40b9-9a10-943231fc7313)
![image6](https://github.com/huggingface/diffusers/assets/58554991/5e824a32-b8ce-4112-8b20-297e158cfebc)
![image7](https://github.com/huggingface/diffusers/assets/58554991/d2aba6df-7e63-4703-99e1-0c17084f3203)
![image8](https://github.com/huggingface/diffusers/assets/58554991/72d648e4-00c1-40e9-b01f-15896feb2639)
![image9](https://github.com/huggingface/diffusers/assets/58554991/cfbd247c-8d36-4edf-9770-c0660263398a)
![image1](https://github.com/huggingface/diffusers/assets/58554991/2983babb-6d93-46b3-aeb8-8f61479af457)

results using EMA weights:
![ema_image0](https://github.com/huggingface/diffusers/assets/58554991/cde35f4d-c1e7-45d4-82f6-83e9e9b08fb2)
![ema_image1](https://github.com/huggingface/diffusers/assets/58554991/4d5a2dfd-aaef-42e9-8d3f-c3339573630d)
![ema_image2](https://github.com/huggingface/diffusers/assets/58554991/0a70def3-1ca1-45da-87f7-b46904a7050f)
![ema_image3](https://github.com/huggingface/diffusers/assets/58554991/b1a3eb6a-4896-4fdc-a940-af1c4bd157d1)
![ema_image4](https://github.com/huggingface/diffusers/assets/58554991/1a792f71-3ca7-40de-83e3-e725edb206d0)
![ema_image5](https://github.com/huggingface/diffusers/assets/58554991/cb0ff7ab-0240-477e-8701-e49c8573d196)
![ema_image6](https://github.com/huggingface/diffusers/assets/58554991/91c1600c-9319-4c1d-91ef-7eae841b62f1)
![ema_image7](https://github.com/huggingface/diffusers/assets/58554991/864681c0-bd77-4ae7-bc37-51f539564b5e)
![ema_image8](https://github.com/huggingface/diffusers/assets/58554991/5cb91ed7-aab2-45cd-9757-323cd05274a1)
![ema_image9](https://github.com/huggingface/diffusers/assets/58554991/e568861e-24de-424c-8a36-e4d3266a12d2)

**Comment by standardAI** [comment]
It seems that the original paper used a learning rate of 4e-04, 800K training iterations didn't use 8-bit Adam and mixed precision. Do these make a difference?

**Comment by sukjunhwang** [comment]
Hi @yhx-03 , could you reproduce the paper results using the code?

**Comment by sukjunhwang** [comment]
Hi @yhx-03 , were you able to reproduce the paper results using the code? I believe iCT also used 400K iterations.
